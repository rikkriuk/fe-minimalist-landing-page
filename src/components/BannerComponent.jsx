import React from "react";
import ThreeHandBannerImg from "../assets/three-hand-banner.png";

const BannerComponent = () => {
   return (
      <section className="relative flex flex-col justify-center items-start md:items-center bg-[#FEC00A] px-6 py-28 mt-48 mb-20">
         <div className="block md:absolute -left-10 -z-0 ">
            <h1 className="text-6xl lg:text-8xl xl:text-9xl font-extrabold text-white mb-10 md:mb-20">dsgnr.</h1>
         </div>

         <div className="flex pb-16 md:pb-3 items-start md:items-center gap-16 flex-col md:flex-row md:justify-center lg:justify-end">
            <h4 className="font-bold w-full md:w-2/5 text-[44px]">Grow Your Business. 
            Build Great Products.</h4>
            <button className="font-semibold text-2xl px-8 py-5 bg-black text-white">
               Book a free call
            </button>
         </div>

         <div className="absolute right-16 bottom-0 z-0">
            <img src={ThreeHandBannerImg} alt="Three hand banner" width={369} />
         </div>
      </section>
   )
}

export default BannerComponent;