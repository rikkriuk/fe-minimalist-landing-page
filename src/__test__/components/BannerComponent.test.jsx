import React from "react";
import { render, screen } from "@testing-library/react";
import BannerComponent from "../../components/BannerComponent";

describe("BannerComponent", () => {
   test("renders the title", () => {
      render(<BannerComponent />);
      expect(screen.getByText("dsgnr.")).toBeInTheDocument();
      expect(screen.getByText("Grow Your Business. Build Great Products.")).toBeInTheDocument();
   });

   test("renders the button", () => {
      render(<BannerComponent />);
      expect(screen.getByText("Book a free call")).toBeInTheDocument();
   })

   test("matches the snapshot", () => {
      const { asFragment } = render(<BannerComponent />);
      expect(asFragment()).toMatchSnapshot();
   });
})