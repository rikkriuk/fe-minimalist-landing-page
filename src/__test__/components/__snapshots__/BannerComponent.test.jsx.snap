// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerComponent matches the snapshot 1`] = `
<DocumentFragment>
  <section
    class="relative flex flex-col justify-center items-start md:items-center bg-[#FEC00A] px-6 py-28 mt-48 mb-20"
  >
    <div
      class="block md:absolute -left-10 -z-0 "
    >
      <h1
        class="text-6xl lg:text-8xl xl:text-9xl font-extrabold text-white mb-10 md:mb-20"
      >
        dsgnr.
      </h1>
    </div>
    <div
      class="flex pb-16 md:pb-3 items-start md:items-center gap-16 flex-col md:flex-row md:justify-center lg:justify-end"
    >
      <h4
        class="font-bold w-full md:w-2/5 text-[44px]"
      >
        Grow Your Business. Build Great Products.
      </h4>
      <button
        class="font-semibold text-2xl px-8 py-5 bg-black text-white"
      >
        Book a free call
      </button>
    </div>
    <div
      class="absolute right-16 bottom-0 z-0"
    >
      <img
        alt="Three hand banner"
        src="[object Object]"
        width="369"
      />
    </div>
  </section>
</DocumentFragment>
`;
